### DISCUSSION TITLE
How does my application figure out if connecting to server failed?

### DISCUSSION THREAD

**Discussion by oyvindlr** [pymodbus-dev/pymodbus#2116] [Help] [answered]
For the synchronous TCP client, whenever "connect()" fails, it prints a message to the console and that's it. This isn't behaviour I would like to see from a library, I would rather it raised an exception or in some other way indicated failure to the caller, so the caller could handle it. Is it possible to configure this behaviour somehow?

**Comment by janiversen** [comment]
No currently not, mainly due to the reconnect feature.

I think if you disable the reconnect feature, then you get an exception.

**Comment by janiversen** [comment]
Apart from that if you want a different behavior then pull requests are welcome.

**Comment by janiversen** [comment, marked as answer]
But connect() returns and if nothing else you can always check with connected(; if a connection is active.

**Comment by oyvindlr** [comment]
Ah, that makes sense. Thanks for the quick replies!
